src/iam/certhandler/modules/pkcs11.cpp

```diff
@@ -0,0 +1,952 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "aos/iam/modules/pkcs11.hpp"
+
+#include "aos/common/os.hpp"
+#include "aos/common/tools/convert.hpp"
+#include "aos/common/tools/ioutils.hpp"
+
+#include "../log.hpp"
+
+namespace aos {
+namespace iam {
+namespace certhandler {
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+PKCS11::PKCS11(const String& certType, const ModuleConfig& config)
+    : mCertType(certType)
+    , mConfig(config)
+{
+}
+
+Error PKCS11::Init(crypto::pkcs11::Context& pkcs11, crypto::x509::ProviderItf& x509Provider)
+{
+    mX509Provider = &x509Provider;
+
+    Error err = crypto::uuid::StringToUUID(String("58ac9ca0-2086-4683-a1b8-ec4bc08e01b6"), cTeeClientUUIDNs);
+    if (!err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
+    err = os::GetEnv(cEnvLoginType, mTeeLoginType);
+    if (!err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
+    if (!mConfig.mUserPINPath.IsEmpty() && mTeeLoginType.IsEmpty()) {
+        return AOS_ERROR_WRAP(Error::Enum::eInvalidConfiguration);
+    }
+
+    err = pkcs11.InitLibrary(mConfig.mLibrary);
+    if (!err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+    mPKCS11 = &pkcs11;
+
+    SetTokenLabel();
```

Comment:
Setting private member in this way decrease code readability IMHO.  To understand that  `SetTokenLabel` actually sets `mTokenLabel`, I have to go inside this function. `mTokenLabel = GetTokenLabel()` explicitly shows that at this place I set private member `mTokenLabel`.